tasks/actorstate/miner.go

```diff
@@ -56,77 +113,383 @@ func (m StorageMinerExtractor) Extract(ctx context.Context, a ActorInfo, node Ac
 		return nil, xerrors.Errorf("loading current miner state: %w", err)
 	}
 
-	minfo, err := curState.Info()
-	if err != nil {
-		return nil, xerrors.Errorf("loading miner info: %w", err)
+	prevState := curState
+	if a.Epoch != 0 {
+		prevActor, err := node.StateGetActor(ctx, a.Address, a.ParentTipSet)
+		if err != nil {
+			return nil, xerrors.Errorf("loading previous miner actor: %w", err)
```

Comment:
context of actor and epoch might be useful

Reply:
agreed, will add.